src/snowflake/snowpark/mock/snowflake_to_pandas_converter.py

```diff
@@ -0,0 +1,218 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
```

Comment:
Why do we need our own copy?

Reply:
it's because the limitation of default behavior of how `pandas.read_csv` parses data and it can not meet the requirement of snowflake data type spec.

e.g. the default `pandas.read_csv` can not validate data of DecimalType() with precision, scale.
also it can not handle timestamp/datetime if the `dtype` of the DataFrame is `object` (which is type we use) and parses those data as raw string.

with those customized converters, we control how to parse data based up on snowflake data type spec.

I have added a comment in the header of this module.